src/main/java/dev/j3fftw/litexpansion/items/PortableCharger.java

```diff
@@ -0,0 +1,224 @@
+            ItemStack borderItem = buildNonInteractable(Material.YELLOW_STAINED_GLASS_PANE, null);
+            ItemStack powerItem = buildNonInteractable(Material.GLOWSTONE, "&4Power");
+
+            // Build and open GUI
+            for (int i = 0; i < 27; i++)
+                inventory.setItem(i, backgroundItem);
+
+            for (int slot : border)
+                inventory.setItem(slot, borderItem);
+
+            inventory.setItem(powerSlot, powerItem);
+            updateSlot(inventory, powerSlot, "&6&lPower Remaining",
+                "&e" + charger.getItemCharge(chargerItem) + "J");
+            inventory.clear(chargeSlot);
+            p.openInventory(inventory);
+
+            // Task that triggers every second
+            new BukkitRunnable(){
+                public void run() {
+
+                    ItemStack deviceItem = inventory.getItem(chargeSlot);
+                    SlimefunItem sfItem = SlimefunItem.getByItem(deviceItem);
+
+                    if (sfItem instanceof Rechargeable && !isItem(deviceItem)) {
+
+                        Rechargeable device = (Rechargeable) sfItem;
+                        float neededCharge = device.getMaxItemCharge(deviceItem)
+                            - device.getItemCharge(deviceItem);
+                        float availableCharge = charger.getItemCharge(chargerItem);
+
+                        // Three different scenarios
+                        if (neededCharge > 0 && availableCharge > 0) {
+
+                            if (neededCharge >= chargeSpeed && availableCharge >= chargeSpeed) {
+                                charger.removeItemCharge(chargerItem, chargeSpeed);
+                                device.addItemCharge(deviceItem, chargeSpeed);
+
+                            } else if (neededCharge < availableCharge) {
+                                charger.removeItemCharge(chargerItem, neededCharge);
+                                device.addItemCharge(deviceItem, neededCharge);
+
+                            } else {
+                                charger.removeItemCharge(chargerItem, availableCharge);
+                                device.addItemCharge(deviceItem, availableCharge);
+                            }
+
+                        } else if (neededCharge == 0) {
+                            Utils.send(p, "&cThis item is already full!");
+
+                        } else {
+                            Utils.send(p, "&cYour charger does not have enough power!");
+                        }
+
+                        // The name of the powerItem NEEDS to be "Portable Charger" to cancel event
+                        updateSlot(inventory, powerSlot, "&6&lPower Remaining",
+                            "&e" + charger.getItemCharge(chargerItem) + "J");
+                    }
+
+                    // Check if GUI is no longer open
+                    if (p.getOpenInventory().getTopInventory() != inventory) {
```

Comment:
Should probably rather check if `getViewers()` is empty, this way we will capture every possible scenario, including Player crashes or leaving since I don't know how this would behave in these edge cases.

It would also be much more reliable as it doesn't rely on a memory address comparison but rather on the actual inventory's data.

Reply:
Wont that only work if its a public inventory?

Reply:
There isn't really such a thing as "public/private", it will work for any inventory.